Initial setup for running the website in Docker

mpetason opened the pull request:
<!-- Thanks for sending a pull request! -->

# Changes

<!-- 
Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! 

- :gift: Add new feature
- :bug: Fix bug
- :broom: Update or clean up current behavior
- :wastebasket: Remove feature or internal logic
-->

- Adds documentation on how to run in Docker

<!--
In addition, categorize the changes you're making using the "/kind" Prow command, example:

/kind <kind>

Supported kinds are: api-change, bug, cleanup, deprecation, removal, documentation, enhancement, performance

-->
/kind documentation

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

<!-- Please include the 'why' behind your changes if no issue exists -->

**Release Note**

<!--
:page_facing_up: If this change has user-visible impact, write a release note in the block
below. Include the string "action required" if additional action is required of
users switching to the new release, for example in case of a breaking change.

Write as if you are speaking to users, not other Knative contributors. If this
change has no user-visible impact, no release-note is needed.
-->
```release-note

```

**Docs**

<!--
:book: If this change has user-visible impact, link to an issue or PR in
https://github.com/knative/docs.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [knative/docs]: <issue or pr link>
- [Feature Track]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

mpetason commented:
/wip

evankanderson commented on docker/README.md:
```diff
@@ -0,0 +1,70 @@
+This will run the service on port 9001, but you can pick any port you want. The container is listening on 1313.
+
+
+Diffs on script files (edited so that mounted volumes aren't deleted or copied over)
```
Suggestion: Rather than putting the script here, why not have a `.patch` file created by something like:
```shell
diff -u ../website/scripts/localbuild.sh localbuild.sh >> diff.patch
diff -u ../website/scripts/processresourcefiles.sh processresourcefiles.sh >> diff.patch
```

This would allow you to use the `patch` command to apply the patch to the original files in the Docker build, and avoid needing to keep an extra copy of the scripts here with only a few lines changed.

evankanderson reviewed (commented):
I'm not sure we have an official place for these sorts of Docker images; most of our images are built during our release process by Prow; I'm not sure what the best option is here.

I _would_ suggest mentioning your current Docker images on DockerHub (assuming that they're public) as an easier way than needing to do your own local `docker build`.

mpetason commented:
Addressed the review requests. Looking for more feedback on where this should live, right now it's website/docker. I don't know if that will cause any issues. It looks like the tests are failing right now due to another issue related to an addition recently.
